### PR TITLE
[kube-prometheus-stack] Added missing enforcedKeepDroppedTargets in the template of prometheus in 0.71.0

### PR DESCRIPTION
**Pull request by Leegin-darknight**
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -425,6 +425,9 @@ spec:
 {{- if .Values.prometheus.prometheusSpec.sampleLimit }}
   sampleLimit: {{ .Values.prometheus.prometheusSpec.sampleLimit }}
 {{- end }}
+{{- if .Values.prometheus.prometheusSpec.enforcedKeepDroppedTargets }}
+  enforcedKeepDroppedTargets: {{ .Values.prometheus.prometheusSpec.enforcedKeepDroppedTargets }}
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml:429]: Please add this to values with an description like this: https://github.com/prometheus-community/helm-charts/blob/1f70a6b763a245f135f34c5a71f7149e3926d84c/charts/kube-prometheus-stack/values.yaml#L3798-L3801

**Review by jkroepke — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml (hunk 1 above)
Comment (Leegin-darknight) [reply charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml:429]: Done. I have added the changes.

File: charts/kube-prometheus-stack/values.yaml [hunk 2]
```diff
@@ -3795,6 +3795,11 @@ prometheus:
     # Set to 'false' to disable global sample_limit. or set to a number to override the default value.
     sampleLimit: false
 
+    #EnforcedKeepDroppedTargetsLimit defines on the number of targets dropped by relabeling that will be kept in memory. 
+    #The value overrides any spec.keepDroppedTargets set by ServiceMonitor, PodMonitor, Probe objects unless spec.keepDroppedTargets 
+    #is greater than zero and less than spec.enforcedKeepDroppedTargets. 0 means no limit.
+    enforcedKeepDroppedTargets: 10
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/values.yaml:3801]: Please use 0 as default value here to avoid a breaking change.

**Review by jkroepke — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/values.yaml (hunk 2 above)
Comment (Leegin-darknight) [reply charts/kube-prometheus-stack/values.yaml]: Done

**Review by jkroepke — APPROVED** [review]
@Leegin-darknight Thanks, it looks good know, however it looks like while you do a merge, the bump of helm chart version is gone now. Please increment the minor version to make the CI happy

**Comment by Leegin-darknight** [comment]
@jkroepke I have bumped up the chart version.

**Comment by jkroepke** [comment]
Please bump
Minor version (5.2.0) an check errors from lint

**Comment by Leegin-darknight** [comment]
> Please bump Minor version (5.2.0) an check errors from lint

Done

**Comment by Leegin-darknight** [comment]
@jkroepke Space was missing in the comments. I have added it. Please review it.  Thanks

**Review by jkroepke — APPROVED** [review]

**Comment by jkroepke** [comment]
Thanks!

**Pull request merged**